### ISSUE TITLE
Is there a way to activeWhen location="somePath" && isLogged = True?

### ISSUE THREAD

**Issue by salvoravida** [single-spa/single-spa-layout#139]
Basically, I 'd like to add another condition to the activateWhen array, but inAND and not in or.

**Comment by joeldenning** [comment]
Perhaps we'll add direct support for conditionals in the layout definition in the future, where routes can be turned on/off via if/else statements. However, there are a couple ways you can do this without waiting for that. See https://github.com/single-spa/single-spa-layout/issues/134 and also the code below:

```js
const applications = constructApplication(...)
const app1 = applications.find(a => a.name === 'app1')
app1.activeWhen = [isLoggedIn, app1.activeWhen]

function isLoggedIn() {
  return isLoggedIn === true
}
```

**Comment by salvoravida** [comment]
Hi, @joeldenning  thank you for your answer.
I'd like to know if activeWhen conditions are mixed with AND or OR.

**Comment by salvoravida** [comment]
activeWhen: string | (location) => boolean | (string | (location) => boolean)[]
Can be a path prefix which will match every URL starting with this path, an activity function (as described in the simple arguments) or an array containing both of them. If any of the criteria is true, it will keep the application active. The path prefix also accepts dynamic values (they must start with ':'), as some paths would receive url params and should still trigger your application. 

@joeldenning your code does not work, as activeWhen mix conditions with OR.

**Comment by joeldenning** [comment]
The array of activeWhen arrays are always OR'ed. If you want to and them, combine them into a single function:

```js
const applications = constructApplication(...)
const app1 = applications.find(a => a.name === 'app1')
const routeActive = app1.activeWhen
app1.activeWhen = function (location) {
  return isLoggedIn === true && routeActive(location)
}
```

Since activity functions are just functions, you can modify them to behave however you'd like

**Comment by salvoravida** [comment]
Yes, but so the declarative routing of spa-layout became useless. You should add conditional rendering to the template. Make sense?

**Comment by joeldenning** [comment]
See my original comment where I said conditionals might be added to the layout definition in a future release of single-spa-layout, and that the code I was sharing was just a workaround. This issue is closed because it's a duplicate of https://github.com/single-spa/single-spa-layout/issues/134, which is tracking that feature.

**Comment by joeframbach** [comment]
I'm a little confused about @joeldenning 's proposed snippet. `routeActive` might not even be a function. It could be a string, or an array.

**Comment by talenW** [comment]
I think the answer is out of date. For conditional `activeWhen` requirement: you can do:
```
const App1 = applications.find(({ name }) => name === 'App1');

App1.activeWhen = App1.activeWhen.map(fn => {
  return location => {
    return isLoggedIn === true && fn(location);
  };
});
```

It can also resolve the requirements that force load a specific application to a registed parent route, like:

```
import {pathToActiveWhen} from 'single-spa';

const App1 = applications.find(({ name }) => name === 'App1');

App1.activeWhen = App1.activeWhen.map(fn => {
  return location => {
    return !pathToActiveWhen('/root/:id/sub-root')(location) && fn(location);
  };
});
```
